### ISSUE TITLE
Upgrade to kotlin 1.4.10

### ISSUE THREAD

**Issue by chineshine** [spring-io/start.spring.io#533]
when I created a spring-boot project with maven , I want to mix kotlin and java in my project . I got the dependecies from start.spring.io which the kotlin version is 1.3.72 , current time is Sept 15,2020. 
I used 'mvn clean package -DskipTests' to package my project, then , I got an issue :

[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.4.0-rc:compile (compile) on project kotlin-sample: Execution compile of goal org.jetbrains.kotlin:kotlin-maven-plugin:1.4.0-rc:compile failed: A required class was missing while executing org.jetbrains.kotlin:kotlin-maven-plugin:1.4.0-rc:compile: kotlin/jvm/internal/KTypeBase

I solved this problem when I upgrade my kotlin'version to 1.4.10.

**Comment by chineshine** [comment]
jdk version is 1.8
and spring-boot version is 2.3.3

**Comment by snicoll** [comment]
Spring Boot `2.3.x` isn't going to move to Kotlin 1.4 so I don't know where that issue comes from or why you believe this is a problem with start.spring.io. 

Can you please clarify what you think we should be changing?

**Comment by chineshine** [comment]
https://github.com/chineshine/kotlin-sample
the link above is the repository of my project 
if you change the kotlin-version to 1.3.72 in the properties tag in pom.xml,  you run the command 'mvn clean package -DskipTests', you will get this issue.
if you change the kotlin-version to 1.4.10,  it will be all right.

**Comment by snicoll** [comment]
Thanks for the sample. Several things to consider:

The POM of your project is not one that is generated by start.spring.io as you moved the parent to a BOM import which, by the way, is not what you should be doing based [on the reference guide](https://docs.spring.io/spring-boot/docs/current/maven-plugin/reference/html/#using-import). Given that the pom has been significantly changed, I don't think an issue here is warranted.

> if you change the kotlin-version to 1.3.72 in the properties tag in pom.xml, you run the command 'mvn clean package -DskipTests', you will get this issue.

I haven't changed anything to your project and running the build leads to:

```
[WARNING]
[WARNING] Some problems were encountered while building the effective model for c.s.sample:kotlin-sample:jar:1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 116, column 21
[WARNING] 'build.plugins.plugin.version' for org.jetbrains.kotlin:kotlin-maven-plugin is missing. @ line 69, column 21
[WARNING] 'build.plugins.plugin.version' for org.springframework.boot:spring-boot-maven-plugin is missing. @ line 64, column 21
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
```

And

```
[WARNING] Runtime JAR files in the classpath have the version 1.3, which is older than the API version 1.4. Consider using the runtime of version 1.4, or pass '-api-version 1.3' explicitly to restrict the available APIs to the runtime of version 1.3. You can also pass '-language-version 1.3' instead, which will restrict not only the APIs to the specified version, but also the language features
[WARNING] /Users/snicoll/.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.3.72/kotlin-reflect-1.3.72.jar: (-1, -1) Runtime JAR file has version 1.3 which is older than required for API version 1.4
[WARNING] /Users/snicoll/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.3.72/kotlin-stdlib-common-1.3.72.jar: (-1, -1) Runtime JAR file has version 1.3 which is older than required for API version 1.4
[WARNING] /Users/snicoll/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-jdk7/1.3.72/kotlin-stdlib-jdk7-1.3.72.jar: (-1, -1) Runtime JAR file has version 1.3 which is older than required for API version 1.4
[WARNING] /Users/snicoll/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-jdk8/1.3.72/kotlin-stdlib-jdk8-1.3.72.jar: (-1, -1) Runtime JAR file has version 1.3 which is older than required for API version 1.4
[WARNING] /Users/snicoll/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.3.72/kotlin-stdlib-1.3.72.jar: (-1, -1) Runtime JAR file has version 1.3 which is older than required for API version 1.4
[WARNING] /Users/snicoll/Downloads/kotlin-sample-master/src/main/kotlin/Adhoc.kt: (42, 16) This class shouldn't be used in Kotlin. Use kotlin.Int instead.
[WARNING] /Users/snicoll/Downloads/kotlin-sample-master/src/main/kotlin/person/person.kt: (12, 13) Variable 'length' is never used
[WARNING] /Users/snicoll/Downloads/kotlin-sample-master/src/main/kotlin/person/person.kt: (54, 17) Parameter 'name' is never used
[WARNING] /Users/snicoll/Downloads/kotlin-sample-master/src/main/kotlin/person/person.kt: (58, 31) Parameter 'age' is never used
[WARNING] /Users/snicoll/Downloads/kotlin-sample-master/src/main/kotlin/person/person.kt: (64, 35) Parameter 'userService' is never used
```

A project built on start.spring.io does not generate those warnings. Because you moved away from the parent, you lost all plugin management as well. The kotlin plugin does not have a version set and Maven warns you about it. 

I've fixed the bom import and added `<version>${kotlin.version}</version>` to the `kotlin-maven-plugin` definition and the problem described does not occur.

All in all, this has nothing to do with start.spring.io. Going forward, if you're looking for support on how to use Maven, please ask on StackOverflow, as mentioned in [the guidelines for contributing](https://github.com/spring-io/initializr/blob/master/CONTRIBUTING.adoc#using-github-issues), we prefer to use GitHub issues only for bugs and enhancements.
